analyses/focal-cn-file-preparation/03-add-cytoband-status-consensus.Rmd

```diff
@@ -0,0 +1,171 @@
+
+```{r}
+scratch_dir <- file.path("..", "..", "scratch")
+```
+
+### Read in files 
+
+The files we are reading in here are the bed files prepared in the shell script using `bedtools coverage`.
+
+```{r message = FALSE}
+all_callable_cytoband_status_df <-
+  read_tsv(file.path(scratch_dir, "intersect_with_cytoband_callable.bed"),
+                  col_names = FALSE)
+
+loss_cytoband_status_df <-
+  read_tsv(file.path(scratch_dir, "intersect_with_cytoband_loss.bed"),
+                  col_names = FALSE)
+
+gain_cytoband_status_df <-
+  read_tsv(file.path(scratch_dir, "intersect_with_cytoband_gain.bed"),
+                  col_names = FALSE)
+```
+
+## Wrangle and merge consensus cytoband status data
+
+### Filter each of the cytoband status data.frames
+
+```{r}
+all_callable_cytoband_status_df <-
+  all_callable_cytoband_status_df %>%
+  select(
+    chr = X1,
+    cytoband = X4,
+    band_length = X8,
+    callable_fraction = X9,
+    Kids_First_Biospecimen_ID = X10
+  ) %>%
+  filter(!is.na(cytoband))
+
+gain_cytoband_status_df <- gain_cytoband_status_df %>%
+  select(
+    chr = X1,
+    cytoband = X4,
+    gain_fraction = X9,
+    Kids_First_Biospecimen_ID = X10
+  ) %>%
+  filter(!is.na(cytoband))
+
+loss_cytoband_status_df <- loss_cytoband_status_df %>%
+  select(
+    chr = X1,
+    cytoband = X4,
+    loss_fraction = X9,
+    Kids_First_Biospecimen_ID = X10
+  ) %>%
+  filter(!is.na(cytoband))
+```
+
+### Join all data.frames together
+
+```{r}
+final_df <- all_callable_cytoband_status_df %>%
+  left_join(gain_cytoband_status_df,
+            by = c("chr", "cytoband", "Kids_First_Biospecimen_ID")) %>%
+  left_join(loss_cytoband_status_df,
+            by = c("chr", "cytoband", "Kids_First_Biospecimen_ID"))
+```
+
+### Add `dominant_status` field to final data.frame
+
+```{r}
+# Create a dominant status column
+final_df <- final_df %>%
+  replace_na(list(
+    gain_fraction = 0,
+    loss_fraction = 0
+  )) %>%
+  mutate(
+    dominant_status = case_when(
+      callable_fraction < 0.5 ~ "uncallable",
+      gain_fraction / callable_fraction > 0.5 ~ "gain",
+      loss_fraction / callable_fraction > 0.5 ~ "loss",
+      (gain_fraction + loss_fraction) / callable_fraction > 0.5 ~ "unstable",
+      TRUE ~ "neutral"
+    )
+  )
+```
+
+### Read in and join original UCSC cytoband data
+
+This step is to define any additional uncallable cytoband regions.
+
+```{r message = FALSE}
+ucsc_cytoband_df <- data.table::fread("http://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/cytoBand.txt.gz", data.table = FALSE)
+
+ucsc_cytoband_df <- ucsc_cytoband_df %>%
+  mutate(band_length = V3 - V2) %>%
+  select(chr = V1, cytoband = V4, band_length) %>%
+  filter(chr %in% paste0("chr", 1:23)) # remove nonstandard chroms.
+
+# Join the cytoband data from the original UCSC cytoband file with the data
+# bedtools coverage files wrangled in the steps above
+final_df <- final_df %>%
+  right_join(ucsc_cytoband_df, by = c("chr", "cytoband", "band_length")) 
```

Comment:
Realizing this whole step may be redundant, since the `bedtools coverage` is including all of the ucsc cytobands already.

Reply:
Agreed, I will remove this section.